relay-server/src/services/buffer/common.rs

```diff
@@ -17,9 +17,22 @@ impl ProjectKeyPair {
         }
     }
 
+    pub fn own_key(&self) -> ProjectKey {
+        self.own_key
+    }
+
+    pub fn sampling_key(&self) -> ProjectKey {
+        self.sampling_key
+    }
+
+    pub fn same_keys(&self) -> bool {
+        self.own_key == self.sampling_key
```

Comment:
nit: nicer semantics?

```suggestion
    pub fn has_distinct_sampling_key(&self) -> bool {
        self.own_key != self.sampling_key
```